src/modules/endpoint/sagas/loadCategories.js

```diff
@@ -17,37 +17,44 @@ function * fetchCategoriesMap (city: string, language: string): Saga<CategoriesM
   return categoriesPayload.data
 }
 
-function * loadCategories (city: string, language: string, dataContainer: DataContainer, shouldUpdate: boolean): Saga<FetchMapType> {
+function * loadCategories (
+  city: string,
+  language: string,
+  dataContainer: DataContainer,
+  shouldUpdate: boolean
+): Saga<FetchMapType> {
+  let categories: CategoriesMapModel
+
   // If data was loaded and should not be updated, return
-  if (dataContainer.categoriesAvailable() && !shouldUpdate) {
-    console.debug('Using cached categories')
-    return {}
-  }
+  if (!dataContainer.categoriesAvailable() || shouldUpdate) {
+    console.debug('Fetching categories')
 
-  console.debug('Fetching categories')
+    // TODO: data was loaded but should be incrementally updated. This will be done in NATIVE-3
 
-  // TODO: data was loaded but should be incrementally updated. This will be done in NATIVE-3
+    const categoriesMap: ?CategoriesMapModel = yield call(fetchCategoriesMap, city, language)
 
-  const categoriesMap: ?CategoriesMapModel = yield call(fetchCategoriesMap, city, language)
+    if (!categoriesMap) {
```

Comment:
Is it possible that `fetchCategoriesMap` returns null or undefined? As far as I can see, it's not possible (or the flow types in `fetchCaegoriesMap` are wrong. So imo this check could be removed.

Reply:
I removed the `?`. Sags is a bit weak on flow types...